### PR TITLE
Add support for Solaris /dev/poll.

### PR DESCRIPTION
**Pull request by psumbera**
https://docs.oracle.com/cd/E19253-01/816-5177/6mbbc4g9n/index.html

### CODE REVIEW

**Comment by psumbera** [comment]
@gnzlbg can you please review?

**Comment by JohnTitor** [comment]
@psumbera Could you rebase to make CI green?
r? @JohnTitor

**Comment by JohnTitor** [comment]
@psumbera It'd be great if you could ping me when updating, otherwise the build log will be deleted and I miss it :/ And I'd prefer rebase to merge here.

File: src/unix/solarish/mod.rs [hunk 1]
```diff
@@ -1845,6 +1864,13 @@ pub const TIOCMGET: ::c_int = tIOC | 29;
 pub const TIOCREMOTE: ::c_int = tIOC | 30;
 pub const TIOCSIGNAL: ::c_int = tIOC | 31;
 
+pub const DP_POLL: ::c_int = 0xd001;
+pub const DP_ISPOLLED: ::c_int = 0xd002;
+#[cfg(target_os = "illumos")]
+pub const DP_PPOLL: ::c_int = 0xd003;
+#[cfg(target_os = "illumos")]
+pub const DP_EPOLLCOMPAT: ::c_int = 0xd004;
```
Comment (pfmooney) [inline src/unix/solarish/mod.rs:1872]: While this does appear in the devpoll header, it's effectively private for the epoll shim in libc and probably isn't useful to expose for `/dev/poll` consumers in rust.

**Comment by JohnTitor** [comment]
I'm going to close this as inactive but feel free to re-open with rebasing and addressing the review from Patrick. Thanks for contributing!

**Pull request closed without merging**